packages/runner/src/context.ts

```diff
@@ -53,6 +53,13 @@ export function withTimeout<T extends (...args: any[]) => any>(
       timer.unref?.()
 
       function resolve(result: unknown) {
+        // if test/hook took too long in microtask, setTimeout won't be triggered,
+        // but we still need to fail the test, see
+        // https://github.com/vitest-dev/vitest/issues/2920
+        if (now() - startTime >= timeout) {
+          reject(new Error(makeTimeoutMsg(isHook, timeout)))
+          return
+        }
         clearTimeout(timer)
```

Comment:
Let's always clear the timeout

Reply:
`reject` calls `clearTimeout` before `reject_` :eyes: 

https://github.com/vitest-dev/vitest/blob/0c38af820b10a9b82ca6f5b7ce256ba265892715/packages/runner/src/context.ts#L60-L63

Reply:
i mean call it before the now check